### PR TITLE
Replace host network mode so it works on Docker Desktop for Mac/Windows as well

### PR DESCRIPTION
**Pull request by jimmygchen**
The host networking driver is only supported on Linux hosts, and not on other platforms (See [docs](https://docs.docker.com/network/host/)). Without host networking, the ports are not exposed to host and ports need to be published for `prometheus` and `grafana` to be accessible. 

I haven't been able to verify this on Linux, but this should work on Linux as well from Docker version `20.10.0` released in December 2020.

### CODE REVIEW

File: scrape-targets/scrape-targets.json [hunk 1]
```diff
@@ -4,8 +4,8 @@
       "job": "nodes"
     },
     "targets": [
-      "localhost:5054",
-      "localhost:5064"
+      "host.docker.internal:5054",
+      "host.docker.internal:5064"
```
Comment (michaelsproul) [inline scrape-targets/scrape-targets.json:8]: I can't get this working on Linux. I tried adding `extra_hosts` to the docker-compose.yaml and changing my metrics listen address to `0.0.0.0` on Lighthouse, but it still times out in Prometheus with this error:

> Get "http://host.docker.internal:5054/metrics": context deadline exceeded

I followed the instructions from here: https://stackoverflow.com/questions/48546124/what-is-linux-equivalent-of-host-docker-internal/67158212#67158212

Comment (michaelsproul) [reply scrape-targets/scrape-targets.json:8]: I'm using Docker 24.0.6, so I don't think it's a version issue. And I can see `172.17.0.1` in `/etc/hosts` inside the container. It just times out when I make requests to it

Comment (jimmygchen) [reply scrape-targets/scrape-targets.json:8]: hmm interesting, I don't have a Linux to test rn but was trying to run this again on my Mac today and reminded me of this branch 😅 

Not sure why but I had to rebuild the images to get Grafana working again for me (the containers run fine but I can't access the UI).. very strange. Although it looks like you're having a different issue. Could it be firewall on your Linux host?

Yeah it's probably best to leave this PR unmerged for now.

Comment (michaelsproul) [reply scrape-targets/scrape-targets.json:8]: I don't have a firewall 😳

Comment (jhlee-young) [reply scrape-targets/scrape-targets.json:8]: @michaelsproul are you trying to use this on Linux to connect Prometheus from outside of the machine? if then, you dont need to disable network_mode `host`, just update `grafana.ini` and `prometheus/Dockerfile`. thats enough

**Comment by AgeManning** [comment]
Bumping this. This still relevant?

**Comment by jimmygchen** [comment]
This may no longer be required with the latest Docker desktop version (4.29) released two weeks ago! I'm going to try this out and will report back.

> Support for host network mode on Docker Desktop for Mac and Windows 

https://www.docker.com/blog/docker-desktop-4-29/#support-host

**Comment by jimmygchen** [comment]
Confirm this now works on Docker Desktop for MacOS version 4.29, with host networking enabled (Beta features)!

Happy to close this now.

**Pull request closed without merging**